include/SH3/system/glbuffer.hpp

```diff
@@ -0,0 +1,112 @@
+/** @file
+ *
+ *  A wrapper and more OO approach to OpenGL Vertex Buffer Objects. This is also helps prevent any
+ *  memory problems as well as providing a whole lot of helper functions for our VBO.
+ *
+ *  @copyright 2017 Palm Studios
+ *
+ *  @date 8-6-2017
+ *
+ *  @author Jesse Buhagiar
+ */
+#ifndef GLBUFFER_HPP_INCLUDED
+#define GLBUFFER_HPP_INCLUDED
+
+#include "GL/glew.h"
+#include "GL/gl.h"
+
+#include <string>
+#include <cstddef>
+
+namespace sh3_gl
+{
+    /**
+     *
+     * An OpenGL Vertex Buffer Object (VBO). Provides a nice, easy, typesafe way
+     * to create and manage OpenGL buffers.
+     */
+    struct sh3_glbuffer final
+    {
+    public:
+
+        enum Target
+        {
+            SH3_ARRAY_BUFFER    = GL_ARRAY_BUFFER,
+            SH3_ELEMENT_BUFFER  = GL_ELEMENT_ARRAY_BUFFER,
+            SH3_TEXTURE_BUFFER  = GL_TEXTURE_BUFFER
+        };
+
+        /**
+         *  Default constructor.
+         */
+        sh3_glbuffer(){Create();buffType = Target::SH3_ARRAY_BUFFER;};
+        sh3_glbuffer(Target type){Create();buffType = type;};
+        sh3_glbuffer(Target type, const std::string& buffname){Create();buffType = type, name = buffname;};
```

Comment:
You could provide default values for those parameters to get rid of the overloads.
Also, is it important that `Create()` is called before initializing the members? Otherwise you can use initializer lists.